src/components/App.js

```diff
@@ -8,33 +8,36 @@ import { CATEGORIES, TASKS } from "../data";
 // console.log({ CATEGORIES, TASKS });
 
 function App() {
-  const [onCategory, setOnCategory] = useState("All")
-  const [onDetails, setOnDetails] = useState("")
-  const [onTasks, setOnTasks] = useState(TASKS)
-  
+  const [onCategory, setOnCategory] = useState("All");
+  const [onTasks, setOnTasks] = useState(TASKS);
 
-  function onTaskFormSubmit(newItem){
-    setOnTasks([...onTasks, newItem])
+  function onTaskFormSubmit(newItem) {
+    setOnTasks([...onTasks, newItem]);
   }
-  
 
-  function handleChange(e){
-    setOnDetails(e.target.value)
+  function handleDelete({ text, category }) {
+    setOnTasks(
+      onTasks.filter((task) => task.text !== text || task.category !== category)
```

Comment:
As we don't have the same object reference, rather just the text and category map under test, that is all we can filter against.

This causes a fun bug that is only introduced due to how the tests make you code in this case